spec/Dynamic-Validator.md

```diff
@@ -0,0 +1,156 @@
+
+## Validator Reward
+The block proposer gets the express fee of the blocks at the end of a term.
+Validators can get the reward after **WITHDRAW_DELAY** terms.
+However, the proposers cannot get all reward if he is not loyal to their duty.
+The reward is decreased according to the rate of the blocks the validator misses to sign.
+TBD: The rate of decreasing.
+
+## Punishment for Validators
+### Downtime
+The validator who doesn't produce blocks is jailed for a while.
+The jailed account cannot be a candidate during **CUSTODY_PERIOD**.
+*SELF_NOMINATE* transaction of the account is rejected.
+Actually, this is not a punishment.
+It is to give validators time to fix the nodes they manage.
+The jailed account can nominate itself again after **CUSTODY_PERIOD**.
+
+### Disloyal Validators
+CodeChain gives a penalty to the validator who doesn't participate in signing the blocks proposed by another node.
+See [Validator Reward](#Validator-Reward) for more information.
+
+### Double Vote
+CodeChain bans the account who double voted.
+The deposit and the reward the criminal earns is slashed and is given to the informant reporting the double vote.
+
+## Transactions
+### SELF_NOMIATION
+* quantity
+* metadata(TBD)
+
+This transaction registers the sender to candidate when the sum of the deposit is larger than **MIN_DEPOSIT**.
+The nomination is valid in **NOMINATE_EXPIRATION**.
+
+The account cannot withdraw the deposit manually.
+It's returned automatically when the account becomes an eligible account.
+
+### WITHDRAW
+* quantity
+
+This transaction withdraws the reward that the node earns as a validator.
+But the validator cannot withdraw the reward before **WITHDRAW_DELAY** passes.
+
+The transaction that tries to withdraw more than the account can fails.
+
+### DELEGATION
+* delegatee
+* quantity
+
+It's a transaction used by the stakeholders to select the validators.
+The validator can delegate the stake as much as it has stakes.
+The validator can delegate to any candidates, validators and jailed accounts.
+The delegations return automatically when the delegatee becomes eligible or banned.
+
+*DELEGATION* transaction to banned or eligible account fails.
+
+
+### REVOCATION
+* delegatee
+* quantity
+
+It's a transaction used by the stakeholders to revoke the delegation.
+The validator can revoke delegations at any time without delay.
+The revoke occurs immediately, but the validator doesn't be ousted before in its term.
+
+The transaction fails when the delegator revokes more than it delegates.
+
+### DOUBLE_VOTE
+* header1
+* sig1
+* header2
+* sig2
+
+This is a transaction to report malicious node.
+The **DOUBLE_VOTE** should be reported in **WITHDRAW_DELAY**.
+The transaction that reports a double vote occurred before **WITHDRAW_DELAY** fails.
+
+The criminal loses all his deposit and rewards and is banned immediately.
+It's the only case that validator set is changed in the term.
```

Comment:
The criminal loses all his deposit and rewards and is banned immediately, and is the only case where a validator set is changed during the term.